src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -441,12 +441,24 @@ private static int GetMediaTypeLength(string input, int startIndex, out MediaTyp
             return current - startIndex;
         }
 
-        private static int GetMediaTypeExpressionLength(string input, int startIndex, out string mediaType)
+        internal static int ParseMediaTypeExpresion<T, K>(
+            string input,
+            int startIndex,
+            out K mediaType,
+            Func<string, int, int, T> typeAndSubtypeFactory,
```

Comment:
Can we make these delegates part of an interface contract that is passed in? The Func doesn't explain what these parameters are meant to do.